slacker/commands/emoji_list_command.py

```diff
@@ -0,0 +1,90 @@
+import os
+import json
+import requests
+
+from argparse import ArgumentParser
+
+from .command import Command
+from slacker.slack_api import SlackAPI
+
+class EmojiListCommand(Command):
+  def name(self):
+    return 'emoji.list'
+
+  def description(self):
+    return 'Lists custom emojis in workspace'
+
+  def aliases(self):
+    return ['emojis']
+
+  def make_parser(self):
+    parser = ArgumentParser(prog = self.name(), description = self.description())
+    parser.add_argument('-p', '--path', type = str,
+                        help='Local file path where to save emojis to')
+    parser.add_argument('-q', '--quiet', action='store_true',
+                        help="Don't print response from Slack API")
+    return parser
+
+  def print_emojis(self, emojis):
+    total = 0
+    for emoji in emojis:
+      print('{:<25} {:<50}'.format(emoji, emojis[emoji]))
+      total += 1
+
+    print('{} custom emojis'.format(total))
+
+  def __download_emojis(self, emojis):
+    save_data = {}
+
+    for emoji in emojis:
+      emoji_url = emojis[emoji]
+      if emoji_url[:5] == 'alias':
+        save_data[emoji] = emoji_url
+        continue
+
+      saved_file_path = self.__download_emoji(emoji, emojis[emoji])
+      save_data[emoji] = saved_file_path
+
+    self.__write_save_file(save_data)
+
+  def __download_emoji(self, emoji_name, url):
+    self.logger.debug('Downloading {}'.format(url))
+
+    res = requests.get(url, stream=True)
+    if res.status_code != 200:
+      self.logger.warning('Unable to download {}: {}'.format(emoji_name, url))
+
+    image_type = res.headers['Content-Type'].split('/')[1]
+    content_length = res.headers['Content-Length']
+
+    emoji_file = '{}.{}'.format(emoji_name, image_type)
+    local_save_path = os.path.join(self.local_save_path, emoji_file)
+
+    self.logger.debug('Writing to disk {}({}) -> {}'.format(url, content_length, self.local_save_path))
+    with open(local_save_path, 'wb') as f:
+      for chunk in res.iter_content(1024):
+        f.write(chunk)
```

Comment:
We should probably implement a `Downloader` class of some kind, or add to `slacker.utility`, at some point to encapsulate this function's functionality so it can be reused for other stuff.

Reply:
Agreed. I think we need to extend the project to have useful utility functions

Reply:
https://github.com/netromdk/slacker/issues/53